### ISSUE TITLE
Error downloading big files

### ISSUE THREAD

**Issue by joanahead** [Dolibarr/dolibarr#14897]
See in Dolibarr from 8 to 12.0.2.

When downloading big files (400mb file makes it fail) uploaded to Dolibarr, for example in user documents folder (htdocs/user/document.php), we get an error on certain Apache servers.

The issue can be easily resolved adding this to htdocs/document.php

```
//start of addition
while (ob_get_level())
 ob_end_flush();
//addition end
//the addition goes just before this line, near the end of the document.php file
readfile($fullpath_original_file_osencoded);
```

As far as we know, that addition does not cause any trouble with other servers and solves the problem for us.

**Comment by eldy** [comment]
This looks strange because there is output  buffering open so adding code to close all opened output buffering should make nothing.
May be an external module hase opened it. Did you installed an external module ?

**Comment by joanahead** [comment]
I coded some plugins myself but I did not use buffering anyway when I coded. Besides that, I don't use any external plugin.
Maybe the explanation is as easy as that. Using readfile function loads the file into the buffer, so a large file loads the buffer up until the server crashes because of a memory limit. Remember it does not happen to me if the file is little.
Here I've found an explanation and example of possible solution: http://teddy.fr/2007/11/28/how-serve-big-files-through-php/

**Comment by eldy** [comment]
May be your php has the global setup output_buffering in php.ini set. Which php version do you have ?

**Comment by joanahead** [comment]
I checked phpinfo()...
PHP  7.2.34
output_buffering 4096

I guess by your question that disabling ouput_buffering could solve this but... should Dolibarr assume that all server will have ouput_buffering off or will deactivate it? Or try to solve issue (that only appears with big files) as I proposed?
I guess the answer is up to you but I am interested in your opinion anyway.

Thanks.

**Comment by eldy** [comment]
Default value for php is output_buffering = 0
This help us to understand why problem occurs on rare situation only. But it's true that it should works even with this setup active.
A fix has been done for v13. Feedback will be welcome to confirm that fix is ok in v13. Fix is 581a3d8808bceae7846f649aaf5d3fa1a247ce81
